### DISCUSSION TITLE
Chat Message Improvements (Buttons + Tooltip; see comments)

### DISCUSSION THREAD

**Discussion by Savantford** [Savantford/foundry-weirdwizard#273] [Polishing and Improvements]
- Idea 1: Collapsible descriptions in chat by default (user setting to not collapse them by default)
  - settings for auto collapse messages
  - settings for auto collapse descriptions
- Idea 2: Include an Perform Activity (Roll Attack) button
  - If targets are selected, those are added to the dialog.
  - The dialog can also have a "Select targets" button (probably just a crosshairs button) that can result in the current targetting behaviour.
  - For AoE effects, we could just have a client option of "placing template auto-targets contained tokens".
- Idea 3: Include Instant Effects' buttons (Roll Damage) on the chat card
- Idea 4: Include Active Effects' buttons (functionality: https://foundryvtt.com/packages/item-effects-to-chat-5e, For items containing Active Effects, links to the embedded effects should be put in chat so that they can be drag-and-dropped on targets/tokens/sheets, maybe with a context menu entry for "Apply to selected tokens")

**Comment by mhilbrunner** [comment]
Suggestion: Clicking on item name/icon rolls it to chat.
1. If the item can roll attack, the chat card has a "Roll attack" button that opens the Roll dialog.
1.1 If targets are selected, those are added to the dialog.
1.2 The dialog can also have a "Select targets" button (probably just a crosshairs button) that can result in the current targetting behaviour.
1.3. For AoE effects, we could just have a client option of "placing template auto-targets contained tokens".
2. If the item can roll damage, the chat card has a "Roll damage" button that opens the Roll dialog.
3. For items containing Active Effects, links to the embedded effects should be put in chat so that they can be drag-and-dropped on targets/tokens/sheets, maybe with a context menu entry for "Apply to selected tokens". See <https://foundryvtt.com/packages/item-effects-to-chat-5e>.
4. We can have shortcuts and context menu entries to open dialogs directly instead of creating a chat card. I think PF/5E use something like shift-click item name to directly roll attack, control-click item name to roll damage?
The above gets rid of all the left menu icons (target, bolt, scroll) in the equipment tab, and all but one (Edit) in the summary tab. IMO the we should replace the latter with a context menu entry.

We should also consider having long item descriptions be collapsible in chat, maybe by default or as a user/client option.
